pyocd/coresight/cortex_m.py

```diff
@@ -127,6 +127,9 @@ def register_name_to_index(reg):
             raise KeyError('cannot find %s core register' % reg)
     return reg
 
+def get_ipsr(context):
```

Comment:
I'm wondering if this should be a method on `DebugContext` rather than a standalone function. This would fit the pyOCD style a little better, I think. The main issue is that you'd need duplicate methods on `Target` and/or `CoreSightTarget` and/or `CortexM`. But then, having these duplicates is also kind of pyOCD-style, currently! (I'm thinking of making `Target` derive from `DebugContext`, btw.)

Reply:
Not so keen on that - you have a perfectly viable "read register" API that doesn't feel like it wants to be worked around.  To make it feel more aligned, I would be more inclined to treat PSR submasks like `cfbp`. Each sub-PSR is readable by name using `MRS` in the instruction set, so would make sense to have them readable via the named register API.